File: backend/agents/legalAgent.py
from ai.openrouter import ask_ai


LEGAL_AGENT_PROMPT = """
You are AgentX Legal Advisor Agent.

Your role is to analyze startup ideas from a legal perspective.

You are an expert in:
- Startup compliance
- Business registration
- Licenses
- Contracts
- Intellectual property
- Data privacy
- Legal risks

Always provide practical startup-focused advice.

Return your answer in structured JSON format:

{
    "business_structure": "",
    "registrations_required": [],
    "licenses_required": [],
    "legal_documents": [],
    "compliance_requirements": [],
    "privacy_requirements": [],
    "legal_risks": [],
    "recommendations": []
}

Do not give fake legal guarantees.
Mention that requirements may vary by country and location.
"""


def generate_legal_analysis(startup_idea):

    response = ask_ai(
        LEGAL_AGENT_PROMPT,
        startup_idea
    )

    return {
        "agent": "Legal Agent",
        "result": response
    }

File: backend/agents/legalAgent.js
const ai=require("../ai/aiClient");

async function legalAgent(startupIdea){

const prompt=`

Generate legal checklist.

Business Registration

Licenses

GST

Trademark

Privacy Policy

Terms

Compliance

Startup:

${startupIdea}

`;

return await ai(prompt);

}

module.exports=legalAgent;